usaspending_api/accounts/models/appropriation_account_balances.py

```diff
@@ -67,26 +64,29 @@ class Meta:
     final_objects = AppropriationAccountBalancesManager()
 
     FINAL_OF_FY_SQL = """
-        WITH submission_and_tai AS (
-            SELECT
-                DISTINCT ON (aab.treasury_account_identifier, FY(s.reporting_period_start))
-                aab.treasury_account_identifier,
-                s.submission_id
-            FROM submission_attributes s
-            JOIN appropriation_account_balances aab
-                  ON (s.submission_id = aab.submission_id)
-            ORDER BY aab.treasury_account_identifier,
-                       FY(s.reporting_period_start),
-                       s.reporting_period_start DESC
+        with submission_and_tai as (
+            select distinct treasury_account_identifier, submission_id from (
+                select distinct on (aab.treasury_account_identifier, s.reporting_fiscal_year)
+                    aab.treasury_account_identifier,
+                    s.submission_id
+                from
+                    submission_attributes s
+                    inner join appropriation_account_balances aab on s.submission_id = aab.submission_id
+                order by
+                    aab.treasury_account_identifier,
+                    s.reporting_fiscal_year,
+                    s.reporting_period_end desc,
+                    s.submission_id desc
+            ) t
         )
-        UPDATE appropriation_account_balances aab
-            SET final_of_fy = true
-            FROM submission_and_tai sat
-            WHERE aab.treasury_account_identifier = sat.treasury_account_identifier AND
-            aab.submission_id = sat.submission_id"""
+        update  appropriation_account_balances
+        set     final_of_fy = ((treasury_account_identifier, submission_id) in
+                    (select treasury_account_identifier, submission_id from submission_and_tai))
+        where   final_of_fy != ((treasury_account_identifier, submission_id) in
+                    (select treasury_account_identifier, submission_id from submission_and_tai));
+    """
 
     @classmethod
     def populate_final_of_fy(cls):
         with connection.cursor() as cursor:
-            cursor.execute("UPDATE appropriation_account_balances SET final_of_fy = false")
```

Comment:
This seems OK to remove. I couldn't think of an example but I wanted to make sure resetting all of the values doesn't handle an edge case to reset `final_of_fy` which need to be set to false

Reply:
Should be ok.  Every row is still examined, but only rows that changed will be updated.  There's still room for improvement by limiting to the FY in question, but the query already runs in less than a second and there's a ticket to get rid of `final_of_fy` so I didn't bother.